### ISSUE TITLE
Error calculating average of tp_bigint

### ISSUE THREAD

**Issue by Novemser** [pingcap/tispark#230]
SQL
```sql
select avg(tp_bigint) from full_data_type_table
```

```bash
Failed with 
TiSpark:		6.422719009976214E14
Spark With JDBC:        7.4281968802033744E16
TiDB:			74281968802033755.5509
```
TiSpark Plan
```bash
== Physical Plan ==
*HashAggregate(keys=[], functions=[sum(sum(tp_bigint#42L)#153L), sum(count(tp_bigint#42L)#154L)])
+- Exchange SinglePartition
   +- *HashAggregate(keys=[], functions=[partial_sum(sum(tp_bigint#42L)#153L), partial_sum(count(tp_bigint#42L)#154L)])
      +- TiDB CoprocessorRDD{[table: full_data_type_table] , Columns: [tp_bigint], Aggregates: Sum([tp_bigint]), Count([tp_bigint])}
```

`6.422719009976214E14` and `7.4281968802033744E16` differs a lot, need to investigate this ASAP.

**Comment by Novemser** [comment]
There's a `Long` overflow in `Cast.java` which caused this problem, will fix this in another PR.

**Comment by birdstorm** [comment]
fixed by #231
